WordPress/gutenberg
# Global styles: add background image to top-level theme.json styles

## Dev Note 📓 

https://github.com/WordPress/gutenberg/pull/59354#issuecomment-2138583222



## What?

Part of:

- https://github.com/WordPress/gutenberg/issues/54336

Enabling background image block supports at the root level of global styles.

### TODO
- [x] Check santization and parsing. Could we assume relative path to the theme and use `get_theme_file_uri( $path )`?
- [x] Ensure styles appear in site editor (New PR?)
- [x] Add UI controls to global styles (top level) https://github.com/WordPress/gutenberg/pull/59454


## How?
Add `style.background` properties as valid top-level styles in theme.json.

## Testing Instructions

Check this branch out and enable emptytheme for ease of testing.

Below are some test theme.jsons. 

Ensure that the `background-*` CSS appears in the  body selector and that everything looks fine. 



```json
{
	"$schema": "../../schemas/json/theme.json",
	"version": 2,
	"styles": {
		"color": {
			"background": "tomato"
		},
		"background": {
			"backgroundImage": {
				"url": "https://i0.wp.com/wordpress.org/files/2023/12/sotw-dotorg-drawer.png?w=1807&ssl=1",
				"source": "file"
			},
			"backgroundPosition": "center center",
			"backgroundRepeat": "repeat",
			"backgroundSize": "cover"
		}
	}
}
```


```json
{
	"$schema": "../../schemas/json/theme.json",
	"version": 2,
	"styles": {
		"color": {
			"background": "tomato"
		},
		"background": {
			"backgroundImage": "url('https://i0.wp.com/wordpress.org/files/2023/12/sotw-dotorg-drawer.png?w=1807&ssl=1')",
			"backgroundPosition": "center center",
			"backgroundRepeat": "no-repeat",
			"backgroundSize": "cover"
		}
	}
}
```

### Testing Instructions for Keyboard


## Screenshots or screencast
